### PR TITLE
Fix stereo sync

### PR DESCRIPTION
**Pull request by servos**
Fix the new spinnaker driver to support camera triggering from another camera.

Mainly this is adding the LineSource, LineSelecetor, and LineMode params.

### CODE REVIEW

File: pointgrey_camera_driver/cfg/PointGrey.cfg [hunk 1]
```diff
@@ -266,24 +266,25 @@ trigger_overlap_modes = gen.enum([gen.const("Off", str_t, "Off", ""),
...
-# gen.add("strobe2_duration", double_t, SensorLevels.RECONFIGURE_RUNNING, "Strobe duration (in milliseconds)", 0.0, 0.0, 1000.0)
+line_sources = gen.enum([gen.const("ExposureActive", str_t, "ExposureActive", ""),
+                         gen.const("ExternalTriggerActive", str_t, "ExternalTriggerActive", "")],
+                         "Line Source")
+
+gen.add("line_source", str_t, SensorLevels.RECONFIGURE_RUNNING, "Line Sources", "ExposureActive", edit_method = line_sources)
+
+gen.add("line_selector", str_t, SensorLevels.RECONFIGURE_RUNNING, "Line Selector", "Line0", edit_method = trigger_sources)
+
+line_mode = gen.enum([gen.const("Input", str_t, "Input", ""),
```
Comment (efernandez) [inline pointgrey_camera_driver/cfg/PointGrey.cfg]: pedantic: `line_modes`

File: pointgrey_camera_driver/src/PointGreyCamera.cpp [hunk 2]
```diff
@@ -130,30 +130,18 @@ bool PointGreyCamera::setNewConfiguration(pointgrey_camera_driver::PointGreyConf
...
-    retVal = setProperty("TriggerMode", std::string("Off"));
-  }
-
+  // Set Trigger and Strobe Settings
+  // NOTE: The trigger must be disabled (i.e. TriggerMode = "Off") in order to configure whether the source is software or hardware.
+  //
+  retVal = setProperty("TriggerMode", std::string("Off"));
+  retVal = setProperty("TriggerSource", config.trigger_source);
+  retVal = setProperty("TriggerSelector", config.trigger_selector);
+  retVal = setProperty("TriggerMode", config.enable_trigger);  // config.enable_trigger
```
Comment (efernandez) [inline pointgrey_camera_driver/src/PointGreyCamera.cpp]: remove leftover comment `  // config.enable_trigger`

File: pointgrey_camera_driver/src/PointGreyCamera.cpp [hunk 3]
```diff
@@ -130,30 +130,18 @@ bool PointGreyCamera::setNewConfiguration(pointgrey_camera_driver::PointGreyConf
...
+
+  ROS_ERROR("Line: %s %s %s", config.line_selector.c_str(), config.line_mode.c_str(), config.line_source.c_str());
```
Comment (efernandez) [inline pointgrey_camera_driver/src/PointGreyCamera.cpp]: Leftover, or testing something?

File: pointgrey_camera_driver/src/PointGreyCamera.cpp [hunk 4]
```diff
@@ -130,30 +130,18 @@ bool PointGreyCamera::setNewConfiguration(pointgrey_camera_driver::PointGreyConf
...
-  }
-  else
-  {
-    // Trigger (Must be switched off to change source!)
-    // (RIGHT CAMERA ONLY)
-    retVal = setProperty("TriggerMode", std::string("Off"));
-    retVal = setProperty("TriggerSource", config.trigger_source);
-    retVal = setProperty("TriggerMode", std::string("On"));
-    retVal = setProperty("TriggerSelector", config.trigger_selector);
-    retVal = setProperty("TriggerActivation", config.trigger_activation_mode);
```
Comment (efernandez) [inline pointgrey_camera_driver/src/PointGreyCamera.cpp:153]: Is `trigger_activation_mode` used now?

**Review by efernandez — APPROVED** [review]
LGTM, only minor comments.

File: pointgrey_camera_driver/src/PointGreyCamera.cpp (hunk 4 above)
Comment (servos) [reply pointgrey_camera_driver/src/PointGreyCamera.cpp:153]: oops, fixed

File: pointgrey_camera_driver/src/PointGreyCamera.cpp [hunk 5]
```diff
@@ -130,30 +130,17 @@ bool PointGreyCamera::setNewConfiguration(pointgrey_camera_driver::PointGreyConf
...
-    retVal = setProperty("TriggerMode", std::string("Off"));
-  }
-
+  // Set Trigger and Strobe Settings
+  // NOTE: The trigger must be disabled (i.e. TriggerMode = "Off") in order to configure whether the source is software or hardware.
+  retVal = setProperty("TriggerMode", std::string("Off"));
+  retVal = setProperty("TriggerSource", config.trigger_source);
+  retVal = setProperty("TriggerSelector", config.trigger_selector);
+  retVal = setProperty("TriggerActivation", config.trigger_activation_mode);
+  retVal = setProperty("TriggerMode", config.enable_trigger);
```
Comment (teyvoniaperception) [inline pointgrey_camera_driver/src/PointGreyCamera.cpp:139]: Do you know if TriggerMode must be enabled before TriggerSelector and TriggerActivation are set? So after the TriggerSource is set having this order instead:
retVal = setProperty("TriggerMode", config.enable_trigger);
retVal = setProperty("TriggerSelector", config.trigger_selector);
retVal = setProperty("TriggerActivation", config.trigger_activation_mode);

**Comment by efernandez** [comment]
:+1:

File: pointgrey_camera_driver/src/PointGreyCamera.cpp (hunk 5 above)
Comment (servos) [reply pointgrey_camera_driver/src/PointGreyCamera.cpp:139]: I'm not sure it matters, seems the it only requires the trigger to be off when setting the source.

**Pull request merged**